sei_dbt/README.md

```diff
@@ -79,17 +108,155 @@ Then, it will:
          path: "path/to/your/database.db"  # Path to your DuckDB database file
          extensions: ['parquet', 'json']   # Optional extensions to load, if needed
             duckdb_project:
+               threads: 4
+         external_root: "data/dev"
 
        test:
          type: duckdb
          path: "test.db"  # Test database path
+         external_root: "data/test" 
          
        prod:
          type: duckdb
          path: "prod.db"  # Production database path
+         external_root: "data/prod"
    ```
+   
+In this `profiles.yml`:
+- We have three environments: `dev`, `test` and `prod`.
+- For each environment, we define:
+  - `external_root`: A variable points to the directory where CSV files or other data are located.  
+  - `dbt-duckdb` only support certain keys in the `profiles.yml`; for example, `external_root`, `settings`, and `config_options`, etc, and each has its own purpose. `custom`, `vars` are not supported. 
+   
+#### Key Usage:
+- **Global Connection Configuration**: You use `profiles.yml` to store environment-specific details such as database credentials, which will be applied to all dbt projects that use this profile.
+- **Sensitive Information**: The `profiles.yml` file is the right place to store sensitive information like database credentials, API tokens, or connection settings. It is generally **not version-controlled**, as it may contain secrets.
+- **Multiple Environments**: You can define multiple environments (like `dev`, `prod`, `test`) within `profiles.yml` and switch between them using the `target` option. This is useful when different teams or environments need to connect to different databases or use different credentials.
+
+#### Default Location:
+- The `profiles.yml` file is located by default in `~/.dbt/profiles.yml`, making it global across multiple dbt projects on your local machine. If you use a Windows machine, it could be in `C:\Users\myusername\.dbt`.
+- You can also override this by explicitly specifying the location of `profiles.yml` with the `DBT_PROFILES_DIR` environment variable.
+
+
+To **infer variables from `profiles.yml`** in your **dbt model file**, you need to use **Jinja** templating and the `target` object, which exposes information from `profiles.yml`.
+
+
+#### Step 2: Access the Variables in the Model File
+
+
+Here’s an example of how you might use the `data_path` variable inside a dbt model:
+
+```sql
+-- models/staging/my_staging_model.sql
+
+WITH raw_data AS (
+
+    -- Use the data_path from profiles.yml
+    SELECT *
+    FROM read_csv_auto('{{ target.external_root }}/my_file.csv')
+
+),
+
+final_data AS (
+
+    SELECT
+        *
+    FROM raw_data
+
+)
+
+SELECT * FROM final_data;
+```
+
+- **`{{ target.vars.data_path }}`**: This Jinja expression accesses the `data_path` variable from `profiles.yml`. When you run dbt with the `--target dev` flag, it will dynamically insert `"data/dev"`, and for `--target prod`, it will insert `"data/prod"`.
```

Comment:
This example does not align with the example just above. 
```suggestion
- **`{{ target.external_root }}`**: This Jinja expression accesses the `external_root` variable from `profiles.yml`. When you run dbt with the `--target dev` flag, it will dynamically insert `"data/dev"`, and for `--target prod`, it will insert `"data/prod"`.
```

Reply:
I have changed it.